app/assets/javascripts/student_profile/pdf/student_profile_pdf.js

```diff
@@ -0,0 +1,195 @@
+(function() {
+  window.shared || (window.shared = {});
+
+  
+  // Returns a list of monthKeys that are within the time window for this chart.
+  function monthKeys(nowMomentUTC, monthsBack) {
+    var lastMonthMomentUTC = nowMomentUTC.clone().date(1);
+    return _.range(monthsBack, -1, -1).map(function(monthsBack) {
+      var monthMomentUTC = lastMonthMomentUTC.clone().subtract(monthsBack, 'months');
+      var monthKey = monthMomentUTC.format('YYYYMMDD');
+      return monthKey;
+    }, this);
+  };
+
+  // A function that grabs a monthKey from an event that is passed in.  Should return
+  // a string in the format YYYYMMDD for the first day of the month.
+  // Used for grouping events on the chart.
+  function defaultMonthKey(event) {
+    return moment.utc(event.occurred_at).date(1).format('YYYYMMDD');
+  };
+
+  // Given a list of monthKeys, map over that to return a list of all events that fall within
+  // that month.
+  function eventsToMonthBuckets(monthKeys, events){
+    var eventsByMonth = _.groupBy(events, defaultMonthKey);
+    return monthKeys.map(function(monthKey) {
+      return eventsByMonth[monthKey] || [];
+    });
+  };
+
+  
+
+  // Returns HighCharts categories map, which describes how to place year captions in relation
+  // to the list of monthKeys.  Returns a map of (index into monthKeys array) -> (caption text)
+  //
+  // Example output: {3: '2014', 15: '2015'}
+  function yearCategories(monthKeys) {
+    var categories = {};
+
+    monthKeys.forEach(function(monthKey, monthKeyIndex) {
+      var monthMomentUTC = moment.utc(monthKey);
+      var isFirstMonthOfYear = (monthMomentUTC.date() === 1 && monthMomentUTC.month() === 0);
+      if (isFirstMonthOfYear) {
+        categories[monthKeyIndex] = yearAxisCaption(monthKey);
+      }
+    }, this);
+
+    return categories;
+  };
+
+  function yearAxisCaption(monthKey) {
+    return moment.utc(monthKey).format('YYYY');
+  };
+
+  function monthAxisCaption(monthKey) {
+    return moment.utc(monthKey).format('MMM');
+  };
```

Comment:
This looks good, and the one more step in https://github.com/studentinsights/studentinsights/issues/880 we can remove the copy-paste duplication here next too! 👍